subliminal/core.py

```diff
@@ -703,3 +713,12 @@ def save_subtitles(video, subtitles, single=False, directory=None, encoding=None
             break
 
     return saved_subtitles
+
+
+def raise_for_status(r):
+    if r.status_code == 304:
+        raise TooManyRequests()
+    elif r.status_code == 503:
+        raise ServiceUnavailable
+    else:
+        r.raise_for_status()
```

Comment:
@Diaoul thanks for reviewing. 
I added TooManyRequests only because of Addi7ed. I will add a custom to the provider then

Just a fyi, I did all this based on OpenSubtitles:
https://github.com/Diaoul/subliminal/blob/develop/subliminal/providers/opensubtitles.py#L268-L294

should we change that method too as it checks 503 and others status codes?